crates/ruma-common/src/room_version_rules.rs

```diff
@@ -204,6 +209,23 @@ pub enum RoomIdFormatVersion {
     V2,
 }
 
+/// The format of [PDU] `auth_events` and `prev_events` for a room version.
+///
+/// [PDU]: https://spec.matrix.org/latest/server-server-api/#pdus
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+pub enum ReferenceFormatVersion {
+    /// `[["$id:server", {"sha256": "base64hash"}]]` format, introduced in room version 1.
+    ///
+    /// [spec]: https://spec.matrix.org/latest/rooms/v1/#event-format
+    V1,
+
+    /// `["$id"]` format, introduced in room version 3.
```

Comment:
For consistency with `EventIdFormatVersion`, this would be clearer:

```suggestion
    /// `["$hash"]` format, introduced in room version 3.
```